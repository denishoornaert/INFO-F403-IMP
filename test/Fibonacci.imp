begin
    read(n) ;
    a := 0 ;
    b := 0 ;
    for n from 0 to n do
        tmp := b ;
        b := a + b ;
        a := tmp ;
    done
    print(b) ;
end
