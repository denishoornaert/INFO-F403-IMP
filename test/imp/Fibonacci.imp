(*
    Algorithme to calcul Fibonacci
*)
begin
    read(n) ;
    a := 0 ;
    b := 0 ;
    for n from 0 to n do (* loop from 0 to n *)
        tmp := b ;
        b := a ;
        a := tmp ;
    done
    print(b) ;
end
