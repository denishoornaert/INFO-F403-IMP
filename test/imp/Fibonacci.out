token: begin    lexical unit: BEGIN
token: read     lexical unit: READ
token: (        lexical unit: LPAREN
token: n        lexical unit: VARNAME
token: )        lexical unit: RPAREN
token: ;        lexical unit: SEMICOLON
token: a        lexical unit: VARNAME
token: :=       lexical unit: ASSIGN
token: 0        lexical unit: NUMBER
token: ;        lexical unit: SEMICOLON
token: b        lexical unit: VARNAME
token: :=       lexical unit: ASSIGN
token: 0        lexical unit: NUMBER
token: ;        lexical unit: SEMICOLON
token: for      lexical unit: FOR
token: n        lexical unit: VARNAME
token: from     lexical unit: FROM
token: 0        lexical unit: NUMBER
token: to       lexical unit: TO
token: n        lexical unit: VARNAME
token: do       lexical unit: DO
token: tmp      lexical unit: VARNAME
token: :=       lexical unit: ASSIGN
token: b        lexical unit: VARNAME
token: ;        lexical unit: SEMICOLON
token: b        lexical unit: VARNAME
token: :=       lexical unit: ASSIGN
token: a        lexical unit: VARNAME
token: ;        lexical unit: SEMICOLON
token: a        lexical unit: VARNAME
token: :=       lexical unit: ASSIGN
token: tmp      lexical unit: VARNAME
token: done     lexical unit: DONE
token: ;        lexical unit: SEMICOLON
token: print    lexical unit: PRINT
token: (        lexical unit: LPAREN
token: b        lexical unit: VARNAME
token: )        lexical unit: RPAREN
token: end      lexical unit: END

Identifiers
a   6
b   7
tmp 9
n   5